### PR TITLE
[DD4hep] Introduce DD4hep Era for Tracker

### PR DESCRIPTION
**Pull request by ianna**
* merge DD and DD4hep `TrackerParameters` producer to allow its use in a `dd4hep` Era
* introduce an Era for `trackerParameters` configuration. That allows us reusing the same RECO geometry configuration rather then its DD4hep duplication
* update geometry dictionaries for all 2021 and 2026 scenarios to allow selecting a `dd4hep` Era for Tracker
* regenerate all 2021 and 2026 scenarios
* simplify test configuration based on an `dd4hep` Era
* fix and cleanup test configurations to be able to run them
* cleanup tests
* move test configurations to a `test/python` directory
* introduce a script to automatically run all the tests in `Geometry/TrackerGeometryBuilder`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
@kpedro88 - please, review

File: Configuration/Geometry/python/GeometryDD4hepExtended2021Reco_cff.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #Tracker
 from RecoTracker.GeometryESProducer.TrackerRecoGeometryESProducer_cfi import *
+from Geometry.TrackerGeometryBuilder.trackerParameters_cff import *
```
Comment (davidlange6) [inline Configuration/Geometry/python/GeometryDD4hepExtended2021Reco_cff.py:14]: not an issue created by this PR - however having parallel cffs (with / without DD4hep) _and_ a dd4hep era is a rather error prone situation.. What ensures that these do not get mixed up?

Comment (ianna) [reply Configuration/Geometry/python/GeometryDD4hepExtended2021Reco_cff.py:14]: agree. The dd4hep-specific configurations should go away as soon as all the sub detectors have era-enabled fragments.

**Comment by kpedro88** [comment]
+upgrade

**Comment by ianna** [comment]
@silviodonato, @dpiparo, @qliphy - please, consider integration asap. @ghugo83's and @bsunanda's imminent PRs depend on this.

**Comment by silviodonato** [comment]
+1

**Pull request merged**